### DISCUSSION TITLE
Restoring metadata: faces are missing

### DISCUSSION THREAD

**Discussion by jean-louis67** [photoprism/photoprism#1842] [Q&A] [answered]
Hello,
I regularly back up metadata this way:

`docker-compose exec -T photoprism photoprism backup -i /var/lib/photoprism/mysql/photoprism_20211223-155202.sql`

1) 2 or 3 months ago I successfully used the backup when rehosting PhotoPrism from one server to another

`docker-compose exec -T photoprism photoprism restore -i /var/lib/photoprism/mysql/photoprism_20211223-155202.sql`

I found absolutely everything that was expected.

2) After a false manipulation on my part, I found myself today in the obligation to restore the metadata.

I noticed after the operation that all the albums, as well as the faces, were missing.

Looking at the help of the restore command, I discovered this command:
`docker-compose exec -T photoprism photoprism restore --albums`

And I get my albums back! Perfect!

On the other hand, I do not see what to do to recover all the work of identifying the faces. The number associated with the 'People' menu is 0
When I look at the contents of the database, the _faces_ table contains 90 rows (I expected 150 or so) and the dates in the _matched_at_, _created_at_, _updated_up_ columns are all from today.

Well, that's not a big deal, I can pretty quickly redo the work of assigning faces.
But, for the next time, I would like to understand if the backup/restore process take into account the faces and/or if I made a handling error.
I use a backup produced 3 days ago with a regularly updated PhotoPrism image

Thank you in advance for your review.

Jean-Louis

**Comment by jean-louis67** [comment, marked as answer]
Okay, I hope no one wasted time with this.
The restore obviously went wrong, not because of a problem in the PhotoPrism APIs, but because of an issue with the database software and the configuration I use.
I ended up deleting the containers and recreating them. By analyzing the logs, I saw that there was a problem with mariadb, and more specifically the Innodb plugin. The explanation is that I had moved the _docker-compose.yml_ file from a local directory to a shared directory. And because of the following setting in _docker-compose.yml_ : 
`- "./database: /var/lib/mysql" # Never remove`
it went badly, for an incorrect mounting problem in the shared directory
I replaced with:
`- "/root/photoprism/database: /var/lib/mysql" # Never remove`
and the problem disappeared.

Once all of these issues are sorted out, the restore goes well and I get all the metadata, including albums and faces.

The restore command that most closely matches my expectations is this (the important difference is -F):
`docker-compose exec -T photoprism photoprism restore -i /var/lib/photoprism/mysql/photoprism_20211223-155202.sql -F`

In short, as we say in French, tout est bien qui finit bien (all's well that ends well)!
Apologies for spamming this thread with an issue that is not inherently a PhotoPrism issue.

I take this opportunity to send all of you, Michael, Theresa, users and contributors my best wishes for 2022!

Jean-Louis

**Reply by graciousgrey** [reply]
>I take this opportunity to send all of you, Michael, Theresa, users and contributors my best wishes for 2022!

Thank you :)

We are very happy to have you in our community! We also wish you all the best for 2022 :tada:
